aiida/manage/manager.py

```diff
@@ -11,209 +11,230 @@
 import asyncio
 import functools
-from typing import TYPE_CHECKING, Any, Optional
+from typing import TYPE_CHECKING, Any, Optional, Union
+from warnings import warn
 
 if TYPE_CHECKING:
     from kiwipy.rmq import RmqThreadCommunicator
     from plumpy.process_comms import RemoteProcessThreadController
 
-    from aiida.backends.manager import BackendManager
     from aiida.engine.daemon.client import DaemonClient
     from aiida.engine.persistence import AiiDAPersister
     from aiida.engine.runners import Runner
     from aiida.manage.configuration.config import Config
     from aiida.manage.configuration.profile import Profile
     from aiida.orm.implementation import Backend
 
-__all__ = ('get_manager', 'reset_manager')
+__all__ = ('get_manager',)
 
+MANAGER: Optional['Manager'] = None
 
-class Manager:
-    """
-    Manager singleton to provide global versions of commonly used profile/settings related objects
-    and methods to facilitate their construction.
 
-    In AiiDA the settings of many objects are tied to options defined in the current profile.  This
-    means that certain objects should be constructed in a way that depends on the profile.  Instead of
-    having disparate parts of the code accessing the profile we put together here the profile and methods
-    to create objects based on the current settings.
+def get_manager() -> 'Manager':
+    """Return the AiiDA global manager instance."""
+    global MANAGER  # pylint: disable=global-statement
+    if MANAGER is None:
+        MANAGER = Manager()
+    return MANAGER
 
-    It is also a useful place to put objects where there can be a single 'global' (per profile) instance.
 
-    Future plans:
-      * reset manager cache when loading a new profile
+class Manager:
+    """Manager singleton for globally loaded resources.
+
+    AiiDA can have the following global resources loaded:
+
+    1. A single configuration object that contains:
+        - Global options overrides
+        - The name of a default profile
+        - A mapping of profile names to their configuration and option overrides
+    2. A single profile object that contains:
+        - The name of the profile
+        - The UUID of the profile
+        - The configuration of the profile, for connecting to storage and processing resources
+        - The option overrides for the profile
+    3. A single storage backend object for the profile, to connect to data storage resources
+    5. A single daemon client object for the profile, to connect to the AiiDA daemon
+    4. A single communicator object for the profile, to connect to the process control resources
+    6. A single process controller object for the profile, which uses the communicator to control process tasks
+    7. A single runner object for the profile, which uses the process controller to start and stop processes
+    8. A single persister object for the profile, which can persist running processes to the profile storage
+
     """
```

Comment:
FYI @giovannipizzi, in terms of terminology, I also wrote this in the `Manager` docstring, to explain the "hierarchy" of config -> profile -> profile specific resources